applications/asset_tracker/src/main.c

```diff
@@ -239,6 +251,58 @@ static void send_button_data_work_fn(struct k_work *work)
 	sensor_data_send(&button_cloud_data);
 }
 
+static void send_modem_at_cmd_work_fn(struct k_work *work)
+{
+	enum at_cmd_state state;
+	int err;
+	struct cloud_channel_data modem_data = {
+		.type = CLOUD_CHANNEL_MODEM
+	};
+	struct cloud_msg msg = {
+		.qos = CLOUD_QOS_AT_MOST_ONCE,
+		.endpoint.type = CLOUD_EP_TOPIC_MSG
+	};
+
+	if (strlen(modem_at_cmd_buff) == 0) {
+		err = -ENOBUFS;
+		state = AT_CMD_ERROR;
+	} else {
+#if defined(CONFIG_AT_CMD)
+		err = at_cmd_write(modem_at_cmd_buff, modem_at_cmd_buff,
+				sizeof(modem_at_cmd_buff), &state);
+#else
+		/* Proper error msg has already been copied to buffer */
+		err = 0;
+#endif
+	}
+
+	if (err) {
+		snprintf(modem_at_cmd_buff, sizeof(modem_at_cmd_buff),
+				"AT CMD error: %d, state %d", err, state);
+	} else if (strlen(modem_at_cmd_buff) == 0) {
+		snprintf(modem_at_cmd_buff, sizeof(modem_at_cmd_buff),
+				"OK\r\n");
+	}
+
+	modem_data.data.buf = modem_at_cmd_buff;
+	modem_data.data.len = strlen(modem_at_cmd_buff);
```

Comment:
I think we need to cap the length at ~2k, as the modem can't handle bigger TLS packets. At least I think it applies to data being sent as well.
@rlubos Do you know?

Reply:
I assume the size limit goes both ways.  Either we have a local err  message as a result of the send failure, a local error message saying the at cmd response is too big, or we send a truncated at cmd response.  If the latter is preferred, what should the max length be, after the overhead?

Reply:
I'd vote for sending an error message indicating response is too big, ie > 2kB. Can't think of much else than the CA certificate readout being bigger than the limit, though there are probably other commands that can produce giant responses.